### PR TITLE
Feature/getjobresultlocal

### PR DESCRIPTION
**Pull request by zfengms**
merge job result locally if enableCloudCombine is set to FALSE

### CODE REVIEW

File: R/jobUtilities.R [hunk 1]
```diff
@@ -171,47 +171,84 @@ getJobResult <- function(jobId) {
...
+        if (job$tasks$succeeded == 0) {
+          stop(sprintf(
+            "all tasks failed for job %s, no result will be avaialble",
+            job$jobId
+          ))
+        }
+      }
+    }
+
     if (metadata$enableCloudCombine == "FALSE") {
```
Comment (brnleehng) [inline R/jobUtilities.R]: Doesn't this error if the metadata is NULL?

File: R/jobUtilities.R [hunk 2]
```diff
@@ -171,47 +171,84 @@ getJobResult <- function(jobId) {
...
+        stop(
+          sprintf(
+            "job %s has failed tasks and error handling is set to 'stop', no result will be avaialble",
+            job$jobId
+          )
+        )
+      } else {
+        if (job$tasks$succeeded == 0) {
+          stop(sprintf(
+            "all tasks failed for job %s, no result will be avaialble",
```
Comment (brnleehng) [inline R/jobUtilities.R]: Typo avaialble

File: R/jobUtilities.R [hunk 3]
```diff
@@ -171,47 +171,84 @@ getJobResult <- function(jobId) {
...
 
   if (!is.null(metadata)) {
+    job <- getJob(jobId, verbose = FALSE)
+
+    if (job$jobState == "active") {
+      stop(sprintf("job %s has not finished yet, please try again later",
+                   job$jobId))
+    } else if (job$jobState != "completed") {
+      stop(sprintf(
+        "job %s is in %s state, no job result is available",
```
Comment (brnleehng) [inline R/jobUtilities.R]: typo

File: R/jobUtilities.R (hunk 2 above)
Comment (brnleehng) [inline R/jobUtilities.R]: typo

File: R/jobUtilities.R [hunk 4]
```diff
@@ -171,47 +171,84 @@ getJobResult <- function(jobId) {
...
-            )
-          )
-        } else {
-          if (job$tasks$succeeded == 0) {
-            stop(sprintf(
-              "all tasks failed for job %s, no result will be avaialble",
-              job$jobId
-            ))
+        # Download the blob to the temporary file
+        rAzureBatch::downloadBlob(
```
Comment (brnleehng) [inline R/jobUtilities.R]: Why are we not merging in the cloud?

File: R/jobUtilities.R [hunk 5]
```diff
@@ -171,47 +171,84 @@ getJobResult <- function(jobId) {
...
+    }
+
     if (metadata$enableCloudCombine == "FALSE") {
-      cat("enalbeCloudCombine is set to FALSE, no job merge result is available",
+      cat("enableCloudCombine is set to FALSE, we will merge job result locally",
           fill = TRUE)
 
-      return()
-    }
+      results <- vector("list", job$tasks$completed)
```
Comment (brnleehng) [inline R/jobUtilities.R]: This doesn't work if the job is still running?

File: R/jobUtilities.R (hunk 4 above)
Comment (brnleehng) [inline R/jobUtilities.R]: No retries?

Comment (zfengms) [reply R/jobUtilities.R]: added retry

File: R/jobUtilities.R (hunk 5 above)
Comment (zfengms) [reply R/jobUtilities.R]: we would error out earlier if the job is still running and will not reach here.

File: R/jobUtilities.R (hunk 4 above)
Comment (zfengms) [reply R/jobUtilities.R]: this is addressed by Pablo

File: R/jobUtilities.R (hunk 2 above)
Comment (zfengms) [reply R/jobUtilities.R]: fixed

File: R/jobUtilities.R (hunk 3 above)
Comment (zfengms) [reply R/jobUtilities.R]: I don't see a typo, which word?

File: R/jobUtilities.R (hunk 2 above)
Comment (zfengms) [reply R/jobUtilities.R]: fixed.

File: R/jobUtilities.R (hunk 1 above)
Comment (zfengms) [reply R/jobUtilities.R]: we have a check for null metadata before this so it should not be null here.

**Pull request merged**